include/circt/Dialect/FIRRTL/Passes.td

```diff
@@ -174,4 +174,29 @@ def BlackBoxReader : Pass<"firrtl-blackbox-reader", "CircuitOp"> {
   let dependentDialects = ["sv::SVDialect"];
 }
 
+def PrefixModules : Pass<"firrtl-prefix-modules", "firrtl::CircuitOp"> {
+  let summary = "Prefixes names of modules and mems in a hiearchy";
+  let description = [{
+    This pass looks for modules annotated with the
+    `NestedPrefixModulesAnnotation` and prefixes the names of all modules
+    instantiated underneath it.  This pass will duplicate a module to rename
+    only the modules instantiated in the hiearchy rooted at the annotated
+    module.  If there are no remaining instances of the original module, it
```

Comment:
Random wording suggestion, instead of:

This pass will duplicate a module to rename  only the modules instantiated in the hiearchy rooted at the annotated   module.  

Maybe something like:

This pass will duplicated modules as necessary to given submodules unique names, ...

also typo hiearchy